src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs

```diff
@@ -43,15 +43,15 @@ static class NativeFormatEnumInfo
                     var handle = field.DefaultValue;
                     unsortedBoxedValues[i] = handle.HandleType switch
                     {
-                        HandleType.ConstantSByteValue => (byte)handle.ToConstantSByteValueHandle(reader).GetConstantSByteValue(reader).Value,
+                        HandleType.ConstantSByteValue => (object)(byte)handle.ToConstantSByteValueHandle(reader).GetConstantSByteValue(reader).Value,
```

Comment:
This was a fun headscratcher.

Reply:
What was going on here?

Reply:
type inference not cooperating

Reply:
Yup, things got widened to uint64 and then boxed as uint64. We need to box before widening.